tests/models/test_models.py

```diff
@@ -511,6 +522,42 @@ def test_incremental_classifier_update_masking_only_during_training(self):
             out_masked = out[:, model.active_units == 0]
             assert torch.all(out_masked == model.mask_value)
 
+    def test_multihead_sizes(self):
+        benchmark = get_fast_benchmark(use_task_labels=True, shuffle=True)
+        model = MultiHeadClassifier(in_features=6)
+        sizes = {}
+        for t, exp in enumerate(benchmark.train_stream):
+            sizes[t] = np.max(exp.classes_in_this_experience) + 1
+            avalanche_model_adaptation(model, exp)
+
+        # Second adaptation should not change anything
+        for t, exp in enumerate(benchmark.train_stream):
+            avalanche_model_adaptation(model, exp)
+
+        for t, s in sizes.items():
+            self.assertEqual(s, model.classifiers[str(t)].classifier.out_features)
+
+    def test_avalanche_adaptation(self):
+        model1 = torch.nn.Sequential(MultiHeadClassifier(in_features=6))
+        benchmark = get_fast_benchmark(use_task_labels=True, shuffle=True)
+        avalanche_model_adaptation(model1, benchmark.train_stream[0])
```

Comment:
the test should check that only the correct modules are updated in this test case

Reply:
I was doing it in a test above, I changed them a bit now I have two tests, one for avalanche_model_adaptation() and one for recursive_adaptation, both test that the correct modules were adapted (through checking that the sizes of the heads do not change when doing 2 adaptations). I removed the first test_multihead_size because it was doing the same.